components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/util/ResponseTypeHandlerUtil.java

```diff
@@ -0,0 +1,617 @@
+
+            String tokenId = UUID.randomUUID().toString();
+            newAccessTokenDO.setTokenId(tokenId);
+            oauthAuthzMsgCtx.addProperty(OAuth2Util.ACCESS_TOKEN_DO, newAccessTokenDO);
+
+            // Persist the access token in database
+            try {
+                OAuthTokenPersistenceFactory.getInstance().getAccessTokenDAO().insertAccessToken(accessToken,
+                        authorizationReqDTO.getConsumerKey(), newAccessTokenDO, existingAccessTokenDO, userStoreDomain);
+                deactivateCurrentAuthorizationCode(newAccessTokenDO.getAuthorizationCode(),
+                        newAccessTokenDO.getTokenId());
+                if (!accessToken.equals(newAccessTokenDO.getAccessToken())) {
+                    // Using latest active token.
+                    accessToken = newAccessTokenDO.getAccessToken();
+                    refreshToken = newAccessTokenDO.getRefreshToken();
+                }
+            } catch (IdentityException e) {
+                throw new IdentityOAuth2Exception(
+                        "Error occurred while storing new access token : " + accessToken, e);
+            }
+            tokenDO = newAccessTokenDO;
+            if (log.isDebugEnabled()) {
+                log.debug("Persisted Access Token for " +
+                        "Client ID : " + authorizationReqDTO.getConsumerKey() +
+                        ", Authorized User : " + authorizationReqDTO.getUser() +
+                        ", Timestamp : " + timestamp +
+                        ", Validity period (s) : " + newAccessTokenDO.getValidityPeriod() +
+                        ", Scope : " + OAuth2Util.buildScopeString(oauthAuthzMsgCtx.getApprovedScope()) +
+                        ", Callback URL : " + authorizationReqDTO.getCallbackUrl() +
+                        ", Token State : " + OAuthConstants.TokenStates.TOKEN_STATE_ACTIVE +
+                        " and User Type : " + OAuthConstants.UserType.APPLICATION_USER);
+            }
+
+            // Add the access token to the cache.
+            if (cacheEnabled) {
+                OAuthCache.getInstance().addToCache(cacheKey, newAccessTokenDO);
+                // Adding AccessTokenDO to improve validation performance
+                OAuthCacheKey accessTokenCacheKey = new OAuthCacheKey(accessToken);
+                OAuthCache.getInstance().addToCache(accessTokenCacheKey, newAccessTokenDO);
+                if (log.isDebugEnabled()) {
+                    log.debug("Access Token was added to OAuthCache for cache key : " + cacheKey.getCacheKeyString());
+                    log.debug("Access Token was added to OAuthCache for cache key : " + accessTokenCacheKey
+                            .getCacheKeyString());
+                }
+            }
+
+        }
+        return  tokenDO;
+    }
+
+    public static AuthzCodeDO generateAuthorizationCode(OAuthAuthzReqMessageContext oauthAuthzMsgCtx, boolean cacheEnabled,
```

Comment:
This method also too long

Reply:
This is an already existing method and I have moved it to the util class.